### PR TITLE
review: feat: Add newline at file end and refactor type calculation

### PR DESCRIPTION
**Pull request by MartinWitt**
A newline has been added to the end of each file by convention. Additionally, the calculate method in DefaultJavaPrettyPrinter has been refactored. The type calculation logic is now cleaner and easier to understand, as unnecessary conditional checks have been removed.

### CODE REVIEW

**Comment by SirYwell** [comment]
This covers a part of #5744 and is a general improvement imo.

File: src/test/java/spoon/support/JavaOutputProcessorTest.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ void testCreateJavaFileAssertFileEncodingChanged(@TempDir File tempDir) throws E
 		Factory factory = launcher.getFactory();
 
 		// use characters encoded differently in ISO-8859-01 and UTFs
-		String code = "class ÈmptyÇlàss {}";
+		String code = "class ÈmptyÇlàss {}"+ System.lineSeparator();
```
Comment (I-Al-Istannen) [inline src/test/java/spoon/support/JavaOutputProcessorTest.java:51]:
```suggestion
		String code = "class ÈmptyÇlàss {}" + System.lineSeparator();
```

File: src/test/java/spoon/test/field/FieldTest.java [hunk 2]
```diff
@@ -216,7 +216,7 @@ public void bugAfterRefactoringImports() {
 				"import java.io.File;\n" +
 				"class A {\n" +
 				"    public static final String separator = File.separator;\n" +
-				"}", klass.toStringWithImports());
+				"}\n", klass.toStringWithImports());
```
Comment (I-Al-Istannen) [inline src/test/java/spoon/test/field/FieldTest.java:219]: Are you sure a unix newline is what we want here?

File: src/test/java/spoon/test/imports/ImportTest.java [hunk 3]
```diff
@@ -1858,7 +1858,7 @@ void testAutoimportConflictingSimpleNames() {
 				"        System.out.println(Locale.HELLO);\n" +
 				"        System.out.println(java.util.Locale.GERMANY);\n" +
 				"    }\n" +
-				"}",
+				"}\n",
```
Comment (I-Al-Istannen) [inline src/test/java/spoon/test/imports/ImportTest.java:1861]: Are you sure a unix newline is what we want here?

File: src/test/java/spoon/test/prettyprinter/DefaultPrettyPrinterTest.java [hunk 4]
```diff
@@ -464,7 +464,7 @@ public void testElseIf() {
 				"        } else if (a == 3) {\n" +
 				"        }\n" +
 				"    }\n" +
-				"}";
+				"}\n";
```
Comment (I-Al-Istannen) [inline src/test/java/spoon/test/prettyprinter/DefaultPrettyPrinterTest.java:467]: Are you sure a unix newline is what we want here?

File: src/test/java/spoon/test/field/FieldTest.java (hunk 2 above)
Comment (MartinWitt) [reply src/test/java/spoon/test/field/FieldTest.java:219]: The `@ExtendWith(LineSeparatorExtension.class)` helps us here. I just wanted to keep the same style in the testcase. We could convert this to a textblock?

**Review by I-Al-Istannen — APPROVED** [review]

**Pull request merged**